### PR TITLE
Versioning update

### PR DESCRIPTION
**Pull request by jmmshn**
# Address a few problems with the CI: 

## Modified `./pytest-docs.yml`
Looks like the docs build has been failing for some time due to upstream package changes.

image

Modified the build pipeline to make sure tests pass

## Automatic Versioning and release 

With the new setup, you should just do a release on GitHub and the release workflow will automatically trigger.
I cannot test this without merging into main and affecting Pypi so I want to make sure this is reviewed.

### CODE REVIEW

**Review by janosh — APPROVED** [review]
LGTM :shipit: 😄

File: pyproject.toml [hunk 1]
```diff
@@ -128,3 +128,7 @@ ignore-words-list = "nd,te,ois"
 
 [tool.mypy]
 no_implicit_optional = false
+
+[tool.versioningit.vcs]
+default-tag = "0.0.1"
+method = "git"
```
Comment (janosh) [inline pyproject.toml:134]: what does this do?

**Comment by jmmshn** [comment]
OK, if I break things on PyPI I promise I'll fix ASAP!

File: pyproject.toml (hunk 1 above)
Comment (jmmshn) [reply pyproject.toml:134]: It uses tags to set the version:
https://pypi.org/project/versioningit/0.3.3/

**Pull request merged**